src/player.c

```diff
@@ -419,28 +435,47 @@ static int play (player_t * const player) {
 					break;
 				}
 
-				const int numChannels = av_get_channel_layout_nb_channels (
-						filteredFrame->channel_layout);
-				const int bps = av_get_bytes_per_sample(filteredFrame->format);
-				ao_play (player->aoDev, (char *) filteredFrame->data[0],
-						filteredFrame->nb_samples * numChannels * bps);
-
-				av_frame_unref (filteredFrame);
+				/*Insert frame for playing*/
+				pthread_mutex_lock (&player->aoplay_lock);
+				STAILQ_INSERT_TAIL (&player->queue_head, frm_element, entry);
+				player->queueSize++;
+				if (player->queueSize >= minQueueSize) {
+						player->forcePause = false;
+						pthread_cond_broadcast (&player->aoplay_cond);
+				}
+				pthread_mutex_unlock (&player->aoplay_lock);
+				filteredFrame=NULL;
 			}
 		}
 
-		const unsigned int songPlayed = av_q2d (player->st->time_base) * (double) pkt.pts;
-		pthread_mutex_lock (&player->lock);
-		player->songPlayed = songPlayed;
-		pthread_mutex_unlock (&player->lock);
-		player->lastTimestamp = pkt.pts;
-
-		av_packet_unref (&pkt);
+		/* insert packet for timestamp */
```

Comment:
I’m not sure this is needed, since frames have a pts as well, so we could use that to update songPlayed. The function av_frame_get_best_effort_timestamp might work as well.